File: code/engine.vc2008/xrManagedEngineLib/GamePersistence.cpp
#include "stdafx.h"
#include "GamePersistence.h"
#include "../xrEngine/IGame_Persistent.h"

XRay::CGamePersistence::CGamePersistence() : NativeObject((IntPtr)g_pGamePersistent)
{
	pNativeObject = g_pGamePersistent;
	AddVirtualMethod("shedule_update");
}

XRay::CGamePersistence::CGamePersistence(IntPtr InNativeObject) : NativeObject(InNativeObject)
{
	CAST_TO_NATIVE_OBJECT(IGame_Persistent, InNativeObject);
	AddVirtualMethod("shedule_update");
}

void XRay::CGamePersistence::shedule_update(u32 interval)
{

}

File: game/gamedata/scripts/GamePersistence.cs
using System;
using System.Collections.Generic;

namespace xrSpectre
{
	public class GamePersistence : XRay.CGamePersistence
	{
		public GamePersistence(IntPtr InNativeObject)
		: base(InNativeObject)
		{
			
		}
		
		public override void shedule_update(UInt32 updateInverval) 
		{
			XRay.Log.Warning("ACTOR UPDATE!");
		}
	}
}